Azure/communication-ui-library-android
# [Bug] Potential memory leak fix for bluetooth pair-unpair scenario

## Purpose

* There is a memory hold while bluetooth is paired and unpaired during a call.
* This memory hold is of bluetooth headset and redux state as well. It is fixed. 
* Sometimes, bluetooth re-pair does not pair the audio device successfully.

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test


* join call
* pair a bluetooth, unpair, pair again, unpair again.
* Leave call
* See memory hold with profiler
* Check successful bluetooth device connection

## What to Check
Verify that the following are valid
* Memory Leak
* Bluetooth connectivity

## Other Information
